documentation/docs/05-misc/02-testing.md

```diff
@@ -2,7 +2,155 @@
+
+To get started, install Vitest (`npm install vitest -D`) and JSDOM (a library that shims DOM APIs, `npm install jsdom -D`). Then adjust your `vite.config.js` such that you can simulate a browser environment for testing Svelte components on the client:
+
+```js
+/// file: vitest.config.js
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+	plugins: [
+		/* ... */
+	],
+	test: {
+		// If you are testing components client-side, you need to setup a DOM environment.
+		// If not all your files should have this environment, you can use a
+		// `// @vitest-environment jsdom` comment at the top of the test files instead.
+		environment: 'jsdom'
+	},
+	// Tell Vitest to prefer the browser condition. Svelte uses a so-called exports map
+	// which loads different code depending on the environment - we want to load the browser environment.
+	resolve: process.env.VITEST
+		? {
+				conditions: ['browser']
+			}
+		: undefined
+});
+```
+
+After that, you can create a test file in which you import the component to test, then interact with it programmatically and write expectations about the results:
+
+```js
+/// file: component.test.js
+import { flushSync, mount, unmount } from "svelte";
+import { expect, test } from "vitest";
+import Component from "./Component.svelte";
+
+test("Component", () => {
+	// Instantiate the component using Svelte's mount API
+	const comp = mount(Component, {
+		target: document.body, // document.body is available thanks to JSDOM
+		props: { initial: 0 },
+	});
+
+	expect(document.body.innerHTML).toBe("<button>0</button>");
+
+	// Click the button, then flush the changes so you can synchronously write expectations
+	document.body.querySelector("button")!.click();
+	flushSync();
+
+	expect(document.body.innerHTML).toBe("<button>1</button>");
+
+	// Remove the component from the DOM
+	unmount(comp);
+});
+
+```
+
+While the process is very straightforward, it is also low level and somewhat brittle - the structure of your component might change and you then may need to adjust your query selectors more often than not. Solutions like [@testing-library/svelte](https://testing-library.com/docs/svelte-testing-library/intro/) can help you facilitate this problem.
+
+### Using runes inside your test files
+
+It is possible to use runes inside your test files. First ensure your bundler knows to route the file through the Svelte compiler before running the test by adding `.svelte` to the filename (e.g `counter.svelte.test.js`). After that, you can use runes inside your tests.
+
+```js
+/// file: counter.svelte.test.js
+import { flushSync, mount, unmount } from 'svelte';
+import { expect, test } from 'vitest';
+
+test('Counter', () => {
```

Comment:
this example tests svelte itself, not custom code. it would be better to show the counter.svelte.js file AND a counter.svelte.test.js file.  (Maybe we need a different example where it actually makes sense to use a rune in the test)

Reply:
You're right about that, but at the same time the example should have as little overhead as possible, which is why I chose to make it self-contained. What about


```js
/// file: doubler.svelte.test.js
import { flushSync, mount, unmount } from 'svelte';
import { expect, test } from 'vitest';
import { doubler } from './doubler.svelte';

test('Doubler', () => {
	let count = $state(0);
	let double = doubler(() => count);

	expect(double.value).toEqual(0);

	count = 5;

	expect(double.value).toEqual(10);
});
```

(and something similar for the `$effect.root` example)

Reply:
should the doubler import be from `'doubler.svelte.ts'` ? 

I don't mind super simple examples as long as it is clear that the test tests application behavior, in this case doubler :+1: